### PR TITLE
External and internal dependency upgrades: monorepo and fixes

### PR DESCRIPTION
**Pull request by chrisbolin**
This PR addresses
#52 (Monorepo)
#53 (Upgrade react-native and react-native-svg)
#54 (Fix demo - work by @sibelius)

53 + 54 are tightly coupled. I chose to include 52 because fixing dependencies would require 4 package publications (all the `victory-*-native` packages), just to be completely obviated when moving to the monorepo.

cc/ @boygirl

### CODE REVIEW

File: lib/components/victory-area.js [hunk 1]
```diff
@@ -0,0 +1,19 @@
+import React from "react";
+import { Dimensions, Platform } from "react-native";
+import { VictoryLabel, VictoryContainer, Area, VictoryClipContainer } from "victory-core-native";
```
Comment (boygirl) [inline lib/components/victory-area.js]: I think these imports all need to be changed

Comment (chrisbolin) [inline lib/components/victory-area.js]: Thanks @boygirl! Sorry about that. I don't know why `npm test` missed that.

**Comment by ryancoughlin** [comment]
Happy to give this a go to test it out. I am using the latest version of RN with: bar, line and segment charts.

**Comment by AlanFoster** [comment]
Following the exact install steps, on a fresh checkout, running with android gives:

```
:react-native-svg:processReleaseJavaRes UP-TO-DATE
:react-native-svg:compileReleaseJavaWithJavac
/Users/admin/Documents/react-tests/victory-native/node_modules/react-native-svg/android/src/main/java/com/horcrux/svg/SvgViewManager.java:14: error: package com.facebook.yoga does not exist
import com.facebook.yoga.YogaMeasureMode;
                        ^
/Users/admin/Documents/react-tests/victory-native/node_modules/react-native-svg/android/src/main/java/com/horcrux/svg/SvgViewManager.java:15: error: package com.facebook.yoga does not exist
import com.facebook.yoga.YogaMeasureFunction;
                        ^
/Users/admin/Documents/react-tests/victory-native/node_modules/react-native-svg/android/src/main/java/com/horcrux/svg/SvgViewManager.java:16: error: package com.facebook.yoga does not exist
import com.facebook.yoga.YogaNodeAPI;
                        ^
/Users/admin/Documents/react-tests/victory-native/node_modules/react-native-svg/android/src/main/java/com/horcrux/svg/SvgViewManager.java:35: error: cannot find symbol
    private static final YogaMeasureFunction MEASURE_FUNCTION = new YogaMeasureFunction() {
                         ^
  symbol:   class YogaMeasureFunction
  location: class SvgViewManager
/Users/admin/Documents/react-tests/victory-native/node_modules/react-native-svg/android/src/main/java/com/horcrux/svg/SvgViewManager.java:35: error: cannot find symbol
    private static final YogaMeasureFunction MEASURE_FUNCTION = new YogaMeasureFunction() {
                                                                    ^
  symbol:   class YogaMeasureFunction
  location: class SvgViewManager
5 errors
:react-native-svg:compileReleaseJavaWithJavac FAILED

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':react-native-svg:compileReleaseJavaWithJavac'.
> Compilation failed; see the compiler error output for details.

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output.

BUILD FAILED

Total time: 3.59 secs
Could not install the app on the device, read the error above for details.
Make sure you have an Android emulator running or a device connected and have
set up your Android development environment:
https://facebook.github.io/react-native/docs/android-setup.html
```

**Comment by sibelius** [comment]
@AlanFoster  you should use RN >= 40 to test this PR

**Comment by sibelius** [comment]
you should also use the latest version of react-native-svg

File: lib/components/victory-area.js (hunk 1 above)
Comment (sibelius) [inline lib/components/victory-area.js]: I've tried to run this branch, `unable to resolve module 'victory-core-native'`

is there anything I can do to help with this PR?

Comment (chrisbolin) [inline lib/components/victory-area.js]: @sibelius sorry about that. i've added the WIP tag on the PR, as it is a little half-baked at the moment

**Comment by RafalWilinski** [comment]
After executing:
```
npm install git://github.com/FormidableLabs/victory-native.git#52-monorepo --save
react-native link react-native-svg
react-native run-ios
```

```
Unhandled promise rejection (rejection id: 2406): UnableToResolveError: Unable to resolve module `victory-core-native` from `/Users/Rafal/ReactNative/RNApp/node_modules/victory-native/lib/components/victory-area.js`: Module does not exist in the module map or in these directories:
  /Users/Rafal/ReactNative/RNApp/node_modules
```

In my `node_modules` directory there's only `victory-core` module, not `victory-core-native`.

**Comment by chrisbolin** [comment]
@RafalWilinski sorry, this PR is current broken and WIP. I'll let you know when I un-break it :P

**Comment by chrisbolin** [comment]
@sibelius @boygirl -- i fixed the problems with the import statements. could you give it a try?

**Comment by boygirl** [comment]
@chrisbolin everything is working and rendering correctly except that none of the labels are rendered. Are you seeing this? I wonder if something changed with the `<Text/>` tag in `react-native-svg`, or maybe I wasn't careful enough with semver in `victory-core`

**Comment by sibelius** [comment]
Labels are not working, but we can fix it in separate PRs

**Comment by ryancoughlin** [comment]
Going to pull this down and try it out. And to confirm: I am rendering `VictoryLabel` with my charts, they won't render with this release?

**Comment by AlanFoster** [comment]
@sibelius thanks for the response!

For context; I am actually running the demo app exactly as described in the read me. If there's a problem related to external depencies - should we add pinned version numbers as part of this PR? 🤔

**Comment by sibelius** [comment]
Try to remove your node_modules and install it again

**Comment by sibelius** [comment]
Something yarn does not work well, try to use npm

**Comment by boygirl** [comment]
@ryancoughlin My plan was to merge this, but not release until after the labels get fixed in a smaller more manageable PR

**Comment by ryancoughlin** [comment]
Cool - good plan! Agreed, wanted to confirm.

**Comment by chrisbolin** [comment]
Merging this PR. Will an issue to fix VictoryLabel, which will be fixed before release.

**Comment by chrisbolin** [comment]
@ryancoughlin @boygirl i'm tracking the VictoryLabel issue in #57

**Comment by kevhuang** [comment]
When can we expect 0.7.0 to be released?

**Comment by chrisbolin** [comment]
Hey @kevhuang! We just need to fix the issue with Victory Label in this release. Ideally that will happen in the next few days.

**Comment by sibelius** [comment]
@kevhuang you can install directly from github

it is working fine, only labels are not good yet

**Comment by ryancoughlin** [comment]
I pulled down the latest release and my line chart won't render anymore. Previously I only had this issue when running in release mode. It now happens in both environments. Few notes:

- Tried stripping down all my other victory elements so I was only rendering a default `VictoryLine` 
- Fresh install of `victory-native` including `react-native-svg` 

`package.json`
```
  "dependencies": {
    "eslint-plugin-react-native": "^2.2.0",
    "lodash": "^4.16.6",
    "moment": "^2.15.2",
    "react": "15.4.2",
    "react-native": "^0.42.0",
    "react-native-config": "^0.1.2",
    "react-native-geocoder": "^0.4.6",
    "react-native-maps": "^0.13.0",
    "react-native-router-flux": "^3.38.0",
    "react-native-splash-screen": "^2.0.0",
    "react-native-svg": "^5.1.5",
    "victory-native": "^0.7.0"
  },
```
Here is my tide chart component that renders the line chart: https://gist.github.com/ryancoughlin/74e7d0909856e520943fb198c0282d59

Thoughts? I've been messing around with some items here, but can't seem to track down what is happening.

**Comment by chrisbolin** [comment]
@ryancoughlin thanks for the heads up on `0.7.0`. i hadn't seen any issues with VictoryLine. It almost looks like all of the data is being plotted with an `x` value of `NaN`?

**Comment by chrisbolin** [comment]
@ryancoughlin and you said this happened with `0.6.0` as well, but only under release mode? Was the behavior exactly the same? (with everything piled up on the left side with `NaN`)

**Comment by ryancoughlin** [comment]
@chrisbolin Yup, it looks like the `y` is working. Guessing by seeing a bunch stacked at various heights. Correct, happened with `0.6.0` only under release mode. Same exact behavior. 

What I didn't do yet is try different data, let me do that to rule out anything on my end.

**Comment by ryancoughlin** [comment]
I created similar JSON data with `time` and `height` keys over at http://beta.json-generator.com/ and I got the same results.

**Comment by boygirl** [comment]
@ryancoughlin @chrisbolin I moved this conversation to a new issue: https://github.com/FormidableLabs/victory-native/issues/63

**Pull request merged**